builtin/providers/google/resource_compute_autoscaler.go

```diff
@@ -249,22 +287,47 @@ func resourceComputeAutoscalerRead(d *schema.ResourceData, meta interface{}) err
 		return err
 	}
 
-	zone := d.Get("zone").(string)
-	scaler, err := config.clientCompute.Autoscalers.Get(
-		project, zone, d.Id()).Do()
+	region, err := getRegion(d, config)
 	if err != nil {
-		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
```

Comment:
The error message appears below if the autoscaler is not found in any of the zones. This error is caused by getRegion rather than not finding the autoscaler.